xrpackage.js

```diff
@@ -1882,8 +1895,8 @@ export class XRPackage extends XRNode {
       });
     }
   }
-  async add(p, reason) {
-    await super.add(p, reason);
+  async add(p, {reason, timeout}) {
```

Comment:
I think every HTML file that does `add()` also needs to be updated to handle this.

Reply:
That's probably in the `xrpackage-site` repository.

Reply:
Additionally, if we change `add`, we should probably also change `remove()` for consistency.

Reply:
I couldn't find anything that uses the `reason` in `xrpackage-site` from the search:

https://github.com/webaverse/xrpackage-site/search?p=3&q=%22add%28%22&unscoped_q=%22add%28%22

do you know if there's some place that definitely uses it so I could confirm my search is working?

Reply:
You might be right; `reason` does seem to be used, but only in the event handler, which luckily is not affected by this change: https://github.com/webaverse/xrpackage-site/blob/493ab1d7e3e31019d555db7ded45001cbf091d4a/edit.js#L891

Does everything continue to work with package loading/remove in `xrpackage-site` with this change?

Reply:
yeah it seems to still work -- I tried serving `xrpackage-site` and `xrpackage` locally and tested it with the service worker in debug mode and I can add and remove packages fine.

I've also added a test in https://github.com/webaverse/xrpackage/pull/111/commits/fac5b5adf0f4dd0d443c6e35591f47d3eddfa65b which uses the new `timeout` option

Reply:
oh I think we do use the `reason` a few times actually, XRPackageEngine itself uses `this.add` and `this.remove` a few times, e.g https://github.com/webaverse/xrpackage/blob/6c15ccf236aee905af8ebe2f4b106dbace3bc1df/xrpackage.js#L1662.

Will update them!

Reply:
> Additionally, if we change add, we should probably also change remove() for consistency.

I've updated it so that `reason` for remove() is in an object, but I don't think we need the timeout because nothing is `async` when removing?

Reply:
Yeah I agree, we probably don't need the timeout on remove.